### PR TITLE
ABCP_7 create layers includes equipment aircraft tracks and location

### PR DESCRIPTION
**Pull request by raelong1220**

### CODE REVIEW

File: bcp/client/src/webpack/components/DataLayer.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import React from 'react';
```
Comment (preid) [inline bcp/client/src/webpack/components/DataLayer.js:1]: Maybe I'm missing something, but what's the point of this component?

File: bcp/client/src/webpack/helper/toGeoJSON.js [hunk 2]
```diff
@@ -0,0 +1,154 @@
+import currentLocationJSON from '../../../../public/currentLocations.json';
```
Comment (preid) [inline bcp/client/src/webpack/helper/toGeoJSON.js]: I'm a bit confused why these .json files are being imported, surely there should be a store where the json files are pulled down from the server.

File: bcp/client/src/webpack/App.js [hunk 3]
```diff
@@ -1,17 +1,17 @@
...
 import {Dropdown} from 'react-bootstrap';
-import Map from './pages/map';
-import List from './pages/list';
+import Map from './pages/Map';
+import List from './pages/List';
 import './less/app.less';
 import {ASSET_MODE} from "./constant";
 import { observer } from "mobx-react";
 import useStores from "./hooks/use-stores";
+import './css/thirdparty/font-awesome/css/font-awesome.css';
```
Comment (preid) [inline bcp/client/src/webpack/App.js]: Can we please not bring thirdparty libs unless we have to. Can we not use https://www.npmjs.com/package/@fortawesome/react-fontawesome instead?

**Review by preid — CHANGES_REQUESTED** [review]
There's some inconsistency in formatting, please run lint with fix or prettier across the code base.

I also strongly feel that the .json data files should not be committed. If they are needed for dev let's create a script to run a backup and copy those files under the dev server root.

File: bcp/client/src/webpack/components/DataLayer.js (hunk 1 above)
Comment (raelong1220) [reply bcp/client/src/webpack/components/DataLayer.js:1]: this is to a GeoJSON layer to show different over layers based on the json data feeded

File: bcp/client/src/webpack/components/DataLayer.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
...
+import {GeoJSON} from "react-leaflet";
+
+const DataLayer = ({data, pointToLayer, onEachFeature, filter}) => {
```
Comment (preid) [inline bcp/client/src/webpack/components/DataLayer.js:4]: I'm still not clear on the purpose of this component, are you forcing a re-render by changing the key?

File: bcp/client/src/webpack/helper/toGeoJSON.js [hunk 5]
```diff
@@ -0,0 +1,209 @@
+import {createSearchText} from './map-style';
+import {AIRCRAFT_TYPE} from '../constant';
+
+export const fetchLocations = (cb) => {
```
Comment (preid) [inline bcp/client/src/webpack/helper/toGeoJSON.js]: Why pass a callback here? Just return the promise.

File: bcp/client/src/webpack/helper/toGeoJSON.js [hunk 6]
```diff
@@ -0,0 +1,209 @@
...
+    cb(equipmentGeoJSON);
+};
+
+export const fetchIncidentGeoJSON = (cb) => {
+    fetch('/incidents.json').then(res => res.json()).then(
+        incidentJSON => {
+            cb(incidentJSON);
+        }
+    );
+};
```
Comment (preid) [inline bcp/client/src/webpack/helper/toGeoJSON.js]: It looks like all the callbacks in this module can be replaced with a returned promise.

**Review by preid — CHANGES_REQUESTED** [review]
There are still .json files in public and now also in dist/client, please remove.

File: bcp/client/src/webpack/components/DataLayer.js (hunk 4 above)
Comment (raelong1220) [reply bcp/client/src/webpack/components/DataLayer.js:4]: yeah, if key is not specified( not changing here), GeoJSON will not update when data changes

**Review by preid — APPROVED** [review]

**Pull request merged**